### PR TITLE
Gcp remote backend sa

### PR DESCRIPTION
**Pull request by rochana-atapattu**
This pull request primarily extends the GCP (Google Cloud Platform) configuration options in the `get-target-config` and `lib` packages, and updates the `setup` action to use the new configuration. The new configuration options allow for the specification of a remote backend for GCP.

Here are the most important changes:

* [`get-target-config/src/run.ts`](diffhunk://#diff-b59f20ac6e750e15e4abdcd9c465c5f327da52e809bd7ee42f93b31d026e84fdR98-R100): Added new configuration options for a GCP remote backend in the `run` function. These include `gcp_remote_backend_service_account`, `gcp_remote_backend_workload_identity_provider`, and `gcp_remote_backend_access_token_scopes`. These changes were made in two places in the `run` function. [[1]](diffhunk://#diff-b59f20ac6e750e15e4abdcd9c465c5f327da52e809bd7ee42f93b31d026e84fdR98-R100) [[2]](diffhunk://#diff-b59f20ac6e750e15e4abdcd9c465c5f327da52e809bd7ee42f93b31d026e84fdR143-R145)
* [`lib/src/index.ts`](diffhunk://#diff-4a6bb1aa13b646dff251f71d1cef651ab13d6b1f366ed3a684ebe5f419ef7770R82-R84): Updated the `JobConfig` and `TargetGroup` objects to include the new GCP remote backend configuration options. [[1]](diffhunk://#diff-4a6bb1aa13b646dff251f71d1cef651ab13d6b1f366ed3a684ebe5f419ef7770R82-R84) [[2]](diffhunk://#diff-4a6bb1aa13b646dff251f71d1cef651ab13d6b1f366ed3a684ebe5f419ef7770R102-R103)
* [`setup/action.yaml`](diffhunk://#diff-b6d267e114ed118850925c30d2eba0d7ef272efd6dc98f9ed1dc0caf3200d894R98-R111): Updated the setup action to use the new GCP remote backend configuration options. If `gcp_remote_backend_service_account` is specified, it uses the `google-github-actions/auth` action to authenticate with GCP, and sets the `GOOGLE_BACKEND_CREDENTIALS` environment variable.

### CODE REVIEW

File: setup/action.yaml [hunk 1]
```diff
@@ -95,6 +95,20 @@ runs:
 
     - uses: suzuki-shunsuke/tfaction/export-aws-secrets-manager@main
 
+    - uses: google-github-actions/auth@55bd3a7c6e2ae7cf1877fd1ccb9d54c0503c457c # v2.1.2
+      id: google-backend-credentials
+      if: steps.target-config.outputs.gcp_remote_backend_service_account != ''
+      with:
+        create_credentials_file: true
```
Comment (suzuki-shunsuke) [inline setup/action.yaml:102]: The default value is `true`, so you don't have to set this explicitly.

https://github.com/google-github-actions/auth?tab=readme-ov-file#inputs-miscellaneous

File: setup/action.yaml [hunk 2]
```diff
@@ -95,6 +95,20 @@ runs:
...
+        workload_identity_provider: ${{ steps.target-config.outputs.gcp_remote_backend_workload_identity_provider != '' && steps.target-config.outputs.gcp_remote_backend_workload_identity_provider || steps.target-config.outputs.gcp_workload_identity_provider }}
+        service_account: ${{ steps.target-config.outputs.gcp_service_account }}
```
Comment (suzuki-shunsuke) [inline setup/action.yaml:104]:
```suggestion
        service_account: ${{ steps.target-config.outputs.gcp_remote_backend_service_account }}
```

File: setup/action.yaml [hunk 3]
```diff
@@ -95,6 +95,20 @@ runs:
...
+        access_token_scopes: ${{ steps.target-config.outputs.gcp_remote_backend_access_token_scopes }}
+        token_format: ${{ steps.target-config.outputs.gcp_remote_backend_access_token_scopes != '' && 'access_token' || '' }}
```
Comment (suzuki-shunsuke) [inline setup/action.yaml:106]: Is `token_format` really necessary for GCS backend?

Comment (suzuki-shunsuke) [inline setup/action.yaml:106]:
```suggestion
```

Are `access_token_scopes` and `token_format` really necessary for GCS backend?

**Comment by suzuki-shunsuke** [comment]
Thank you for your contribution!

**Comment by suzuki-shunsuke** [comment]
This repository requires signing commits.

https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

Your commits aren't signed.

image

File: lib/src/index.ts [hunk 4]
```diff
@@ -79,6 +79,9 @@ const JobConfig = z.object({
   gcp_service_account: z.optional(z.string()),
   gcp_workload_identity_provider: z.optional(z.string()),
   gcp_access_token_scopes: z.optional(z.string()),
+  gcp_remote_backend_service_account: z.optional(z.string()),
+  gcp_remote_backend_workload_identity_provider: z.optional(z.string()),
+  gcp_remote_backend_access_token_scopes: z.optional(z.string()),
```
Comment (suzuki-shunsuke) [inline lib/src/index.ts:84]:
```suggestion
```

Do we need to configure access_token_scopes?

File: setup/action.yaml [hunk 5]
```diff
@@ -95,6 +95,20 @@ runs:
...
+
+    - shell: bash
+      if: steps.target-config.outputs.gcp_remote_backend_service_account != ''
+      run: echo "GOOGLE_BACKEND_CREDENTIALS=${{ steps.google-backend-credentials.outputs.credentials_file_path }}" >> "$GITHUB_ENV"
```
Comment (suzuki-shunsuke) [inline setup/action.yaml:110]:
```suggestion
      run: echo "GOOGLE_BACKEND_CREDENTIALS=$GOOGLE_BACKEND_CREDENTIALS" >> "$GITHUB_ENV"
      env:
        GOOGLE_BACKEND_CREDENTIALS: ${{ steps.google-backend-credentials.outputs.credentials_file_path }}
```

**Comment by rochana-atapattu** [comment]
closing this PR. #1643 resolves the requested changes

**Pull request closed without merging**